honeycombio/libhoney-py
# maint: dockerize example and run in ci

## Which problem is this PR solving?

- closes #116 

## Short description of the changes

- add dockerfile (FactorialDockerfile) and docker-compose to run factorial example app in docker
- add makefile with common targets including a spin-up and spin-down of the example app in docker
- add "smoke test" to CI, which right now is just the spin-up and spin-down of the example app

## How to verify this has the expected result

run the various make commands locally and/or [see the results in circle](https://app.circleci.com/pipelines/github/honeycombio/libhoney-py/1230/workflows/46c5f941-4d3d-4d82-8738-9a73809752e6/jobs/9690), with printed output of what would get sent to honeycomb (if a proper api key and endpoint were set instead of test values)

see smoke tests fail with bad code change [as seen here in circle](https://app.circleci.com/pipelines/github/honeycombio/libhoney-py/1233/workflows/5fcc8366-b464-4b10-b43e-cae860df8815/jobs/9700)
